docs/scripts/generate_gallery_text.py

```diff
@@ -55,21 +55,19 @@ def regenerate_gallery():
     gallery_directive_items = []
     with sync_playwright() as p:
         # Generate our browser to visit pages and generate images
-        browser = p.chromium.launch()
+        for ii in range(3):
+            try:
+                browser = p.chromium.launch()
+                break
+            except TimeoutError:
+                print(f"Browser start timed out. Trying again (attempt {ii+2}/3)")
         page = browser.new_page()
 
         for item in track(gallery_items, description="Generating screenshots..."):
             item["id"] = item["name"].lower().replace(" ", "_")
             screenshot = gallery_dir / f"{item['id']}.png"
-
-            for ii in range(3):
-                try:
-                    page.goto(item["website"])
-                    page.screenshot(path=screenshot)
-                    break
-                except TimeoutError:
-                    print(f"{item['name']} timed out. Trying again (attempt {ii+2}/3)")
-                    continue
+            page.goto(item["website"])
```

Comment:
why did you removed the `try.. except` statement from here ? The browser is only launched once so when there were timeout error it was related to the page not the browser launch. By removing it from here if one page doesn't load we'll simply get an error.

Reply:
Maybe i misunderstood something. When I was looking at the readthedocs logs, the timeout always seemed to be on the browser launch action, not on the page visit action. That led me to believe it was the browser launch that we wanted in the try block. But maybe we need both? In your testing did it also time out on the page visit action?

Reply:
on my side I was only getting timeout on the MegEngine page, so yes the timeout was related to pages. So if you experienced the same on launch then yest we need both

Reply:
Ah sounds good - OK the latest commit adds back in the timeout check for pages as well as the browser. It still keeps the browser generation step outside of the loop, since this cuts down on the build time by a bit, and reduces the chance of extra timeouts.